service/src/main/kotlin/xyz/lebkuchenfm/api/auth/ValidateAuthHandler.kt

```diff
@@ -18,74 +15,73 @@ import xyz.lebkuchenfm.domain.users.SetPasswordError
+                            status = HttpStatusCode.Unauthorized,
+                        )
+                    }
 
-                    AuthError.UserDoesNotExistError -> ProblemResponse(
-                        "Could not authenticate.",
-                        "User does not exist.",
-                        status,
-                        instance,
-                    )
+                    AuthError.UserDoesNotExistError -> {
+                        call.respondWithProblem(
+                            title = "Could not authenticate.",
+                            detail = "User does not exist.",
+                            status = HttpStatusCode.Unauthorized,
+                        )
+                    }
 
                     is AuthError.CannotAddNewUserError -> {
-                        val title = "Could not create new user."
-                        val detail = when (error.addNewUserError) {
-                            AddNewUserError.UserAlreadyExists -> "User already exists."
-                            AddNewUserError.UnknownError -> "Something went wrong."
-                        }
-                        ProblemResponse(title, detail, status, instance)
+                        call.respondWithProblem(
+                            title = "Could not create new user.",
+                            detail = when (error.addNewUserError) {
+                                AddNewUserError.UserAlreadyExists -> "User already exists."
+                                AddNewUserError.UnknownError -> "Something went wrong."
+                            },
+                            status = HttpStatusCode.Unauthorized,
+                        )
                     }
 
                     is AuthError.CannotSetPasswordError -> {
-                        val title = "Could not set password for user."
-                        val detail = when (error.setPasswordError) {
-                            SetPasswordError.UserDoesNotExist -> "User does not exist."
-                            is SetPasswordError.ValidationError -> listOfNotNull(
-                                "Password is too weak:",
-                                "too short".takeIf { error.setPasswordError.tooShort },
-                            ).joinToString(separator = " ")
-                            SetPasswordError.UnknownError -> "Something went wrong."
-                        }
-                        ProblemResponse(title, detail, status, instance)
+                        call.respondWithProblem(
+                            title = "Could not set password for user.",
+                            detail = when (error.setPasswordError) {
+                                SetPasswordError.UserDoesNotExist -> "User does not exist."
+                                is SetPasswordError.ValidationError -> listOfNotNull(
+                                    "Password is too weak:",
```

Comment:
I truly dislike any password requirements.
IMO one and only requirement should be that it must exists so `len(password) > 0`

Reply:
I don't agree.